### PR TITLE
add array index checks  #6866

### PR DESCRIPTION
**Pull request by maddin200**

### CODE REVIEW

File: vtkCurveGenerator.cxx [hunk 1]
```diff
@@ -932,7 +932,11 @@ void vtkCurveGenerator::SortByMinimumSpanningTreePosition(vtkPoints* points, vtk
       }
 
     // Add the picked vertex to the MST Set
-    mstSet[nextPointIndex] = true;
+    if (nextPointIndex >= 0 && nextPointIndex < numberOfPoints)
++   {
```
Comment (lassoan) [inline vtkCurveGenerator.cxx:936]:
```suggestion
+     {
```

File: vtkCurveGenerator.cxx [hunk 2]
```diff
@@ -932,7 +932,11 @@ void vtkCurveGenerator::SortByMinimumSpanningTreePosition(vtkPoints* points, vtk
...
++      // Add the picked vertex to the MST Set
++      mstSet[nextPointIndex] = true;
++   }
```
Comment (lassoan) [inline vtkCurveGenerator.cxx:939]:
```suggestion
+     }
```

**Review by lassoan — APPROVED** [review]
Thank you, it looks good (except the alignment of braces that I'll fix)

**Comment by jamesobutler** [comment]
@lassoan Do you want to still integrate this proposed change?

**Pull request closed without merging**